### PR TITLE
Doc and board fixes for CONFIG_BT_DEBUG_MONITOR

### PR DESCRIPTION
**Pull request by carlescufi**
Enable the monitor in all nRF-based boards and document the use of the Kconfig option.

### CODE REVIEW

File: doc/guides/bluetooth/bluetooth-dev.rst [hunk 1]
```diff
@@ -42,6 +42,33 @@ To start developing using this setup follow the :ref:`Getting Started Guide
...
+protocol that interleaves both normal log messages as well as the HCI traffic.
+Set the following Kconfig options to enable this protocol before building your
+application:
+
+.. code-block:: console
+
+   CONFIG_BT_DEBUG_MONITOR=y
+   CONFIG_UART_CONSOLE=n
+
+The first item replaces the :opt:`CONFIG_BT_DEBUG_LOG` option, and the second
```
Comment (jhedberg) [inline doc/guides/bluetooth/bluetooth-dev.rst]: I think this one needs to be `:option:` (that's what CI complains about)

Comment (carlescufi) [reply doc/guides/bluetooth/bluetooth-dev.rst]: Fixed, thanks!

**Review by jhedberg — APPROVED** [review]

File: doc/guides/bluetooth/bluetooth-dev.rst [hunk 2]
```diff
@@ -42,6 +42,33 @@ To start developing using this setup follow the :ref:`Getting Started Guide
...
+protocol that interleaves both normal log messages as well as the HCI traffic.
+Set the following Kconfig options to enable this protocol before building your
+application:
+
+.. code-block:: console
+
+   CONFIG_BT_DEBUG_MONITOR=y
+   CONFIG_UART_CONSOLE=n
+
+The first item replaces the :option:`CONFIG_BT_DEBUG_LOG` option, and the second
```
Comment (dbkinder) [inline doc/guides/bluetooth/bluetooth-dev.rst]: How about this instead, so we get links to the configuration option documentation:
```
Setting :option:`CONFIG_BT_DEBUG_MONITOR` to ``y`` replaces the
:option:`CONFIG_BT_DEBUG_LOG` option, and setting
:option:`CONFIG_UART_CONSOLE` to ``n`` disables the
default ``printk``/``printf`` hooks.

```

**Review by dbkinder — CHANGES_REQUESTED** [review]
a little tweak...

File: doc/guides/bluetooth/bluetooth-dev.rst (hunk 2 above)
Comment (carlescufi) [reply doc/guides/bluetooth/bluetooth-dev.rst]: Done, thank you!

**Review by dbkinder — APPROVED** [review]
Thanks.

**Pull request merged**